keyboards/lily58/keymaps/drasbeck/keymap.c

```diff
@@ -0,0 +1,188 @@
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,                     KC_HOME, KC_PGDN, KC_PGUP, KC_END,  KC_NO,   KC_NUHS, \
+  KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_RO,   KC_ENT, \
+                             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS \
+  ),
+
+  [_RAISE] = LAYOUT( \
+  KC_TRNS,    RALT(KC_1), RALT(KC_2), RALT(KC_3), RALT(KC_4), RALT(KC_5),                         RALT(KC_6),    RALT(KC_7),    RALT(KC_8),    RALT(KC_9),    RALT(KC_0),    KC_DEL, \
+  KC_TRNS,    KC_NO,      KC_NO,      KC_NO,      KC_NO,      KC_NO,                              KC_NO,         KC_NO,         KC_NO,         KC_MINS,       KC_EQL,        KC_RBRC, \
+  KC_ESC,     KC_NO,      KC_NO,      KC_NO,      KC_NO,      KC_NO,                              KC_LEFT,       KC_DOWN,       KC_UP,         KC_RGHT,       KC_NO,         KC_NUHS, \
+  KC_LSFT,    KC_NO,      KC_NO,      KC_NO,      KC_NO,      KC_NO,      KC_NO,      KC_NO,      KC_NO,         KC_VOLD,       KC_VOLU,       KC_MUTE,       KC_RO,         KC_SFTENT, \
+                          KC_RALT,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    RALT(KC_NUBS), KC_TRNS \
+  ),
+
+  [_ADJUST] = LAYOUT( \
+  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, \
+  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, \
+                             _______, _______, _______, _______, _______,  _______, _______, _______ \
+  )
+};
+
+int RGB_current_mode;
+
+// Setting ADJUST layer RGB back to default
+void update_tri_layer_RGB(uint8_t layer1, uint8_t layer2, uint8_t layer3) {
+  if (IS_LAYER_ON(layer1) && IS_LAYER_ON(layer2)) {
+    layer_on(layer3);
+  } else {
+    layer_off(layer3);
+  }
+}
+
+void matrix_init_user(void) {
+    #ifdef RGBLIGHT_ENABLE
+      RGB_current_mode = rgblight_config.mode;
+    #endif
+}
+
+//SSD1306 OLED update loop, make sure to enable OLED_DRIVER_ENABLE=yes in rules.mk
+#ifdef OLED_DRIVER_ENABLE
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+  if (!is_keyboard_master())
+    return OLED_ROTATION_180;  // flips the display 180 degrees if offhand
+  return rotation;
+}
+
+// When you add source files to SRC in rules.mk, you can use functions.
+const char *read_layer_state(void);
+const char *read_logo(void);
+void set_keylog(uint16_t keycode, keyrecord_t *record);
+const char *read_keylog(void);
+const char *read_keylogs(void);
```

Comment:
Just a heads up, if you use a lot of these, you may want to convert to use `oled_write_P` and `oled_write_ln_P` instead, otherwise you may run into issues with using too much memory.